Include freetype source from openjdk

keithc-ca opened the pull request:
* same source as included in jdk11+
* remove `--with-freetype-src` configure option
* `--with-free-type=` accepts only `bundled` or `system`, consistent with other versions

Must be merge concurrently with openj9#TDB.

keithc-ca commented:
jenkins test sanity aix,alinux64,osx,win,win32 jdk8 depends https://github.com/eclipse-openj9/openj9/pull/16182

pshipton commented:
We'll need to modify https://github.com/ibmruntimes/temurin-build/ concurrently as well or the Semeru builds will break.

pshipton commented:
Like jdk11+, we can use `--skip-freetype`, in ibm/build-farm/platform-specific-configurations/windows.sh and mac.sh, or set it as default for all platforms.

pshipton commented on jdk/THIRD_PARTY_README:
```diff
@@ -3315,4 +3315,51 @@ claims in such list are, in fact, Essential Claims.]
 
 --- end of LICENSE ---
 
+%% This notice is provided with respect to the FreeType 2 font engine
```
We're missing a line of dashes to separate this from other licenses, as is done above.
I think we should include the full version here (v2.12.1), similarly to the freetype.md in jdk11. The version isn't included anywhere else.

keithc-ca commented:
Updated `jdk/THIRD_PARTY_README`.

keithc-ca commented:
> We'll need to modify https://github.com/ibmruntimes/temurin-build/ concurrently as well or the Semeru builds will break.

See these pull requests:
* https://github.com/adoptium/temurin-build/pull/3133
* https://github.com/ibmruntimes/temurin-build/pull/28

pshipton commented:
Ok. Now we need to wait for the 0.35 build to be finalized so these changes don't break those builds.

gdams commented:
@keithc-ca do you intend to upstream this to mainstream OpenJDK8 development? It's something the updates project have been hoping to do for some time.

keithc-ca commented:
> do you intend to upstream this to mainstream OpenJDK8 development?

It would seem a reasonable course of action; I'll see if I can find some help to make that happen.

gdams commented:
> It would seem a reasonable course of action; I'll see if I can find some help to make that happen.

Sounds good, more than happy to help @keithc-ca

keithc-ca commented:
Rebased and updated to reflect licence changes in https://github.com/ibmruntimes/openj9-openjdk-jdk11/commit/1f8e80709a0050bf7b49e6069b73a14d4174ed3e.

andrew-m-leonard commented:
What version of Freetype is this src ?

pshipton commented:
> What version of Freetype is this src ?

2.12.1
See the update to THIRD_PARTY_README

gnu-andrew commented:
> > It would seem a reasonable course of action; I'll see if I can find some help to make that happen.
> 
> Sounds good, more than happy to help @keithc-ca

It would definitely be good to see this upstream and should allow us to remove the need to build FreeType within the GitHub Actions for Windows.

I think this would primarily be a backport of [JDK-8193017](https://bugs.openjdk.org/browse/JDK-8193017), though we can reference further bugs to note that we also have updates to later versions. A PR with the subject `"Backport 58ff4ee8c4ec399528338b59eadea996110366a0"` against openjdk/jdk8u-dev should start the ball rolling; see the [SKARA backport process](https://wiki.openjdk.org/display/SKARA/Backports#Backports-BackportPullRequests) for details.

gdams commented:
> > > It would seem a reasonable course of action; I'll see if I can find some help to make that happen.
> > 
> > 
> > Sounds good, more than happy to help @keithc-ca
> 
> It would definitely be good to see this upstream and should allow us to remove the need to build FreeType within the GitHub Actions for Windows.
> 
> I think this would primarily be a backport of [JDK-8193017](https://bugs.openjdk.org/browse/JDK-8193017), though we can reference further bugs to note that we also have updates to later versions. A PR with the subject `"Backport 58ff4ee8c4ec399528338b59eadea996110366a0"` against openjdk/jdk8u-dev should start the ball rolling; see the [SKARA backport process](https://wiki.openjdk.org/display/SKARA/Backports#Backports-BackportPullRequests) for details.

As nobody has picked this up yet I'll have a go at the backport process to jdk8u-dev
